pkg/storage/storage.go

```diff
@@ -446,20 +489,80 @@ func (s *storageModule) createSubvol(size uint64, name string, poolType pkg.Devi
 		return nil, pkg.ErrInvalidDeviceType{DeviceType: poolType}
 	}
 
-	type Candidate struct {
-		Pool      filesystem.Pool
-		Available uint64
+	// Look for candidates in mounted pools first
+	candidates, err := s.checkForCandidates(size, poolType, true)
+	if err != nil {
+		log.Error().Err(err).Msgf("failed to search conditation on unmounted pools")
+	}
+
+	log.Debug().Msgf("Found %d candidates in mounted pools", len(candidates))
+
+	// If no candidates or found in mounted pools, we check the unmounted pools and get the first one that fits
+	if len(candidates) == 0 {
+		log.Debug().Msg("Checking unmounted pools")
+		candidates, err = s.checkForCandidates(size, poolType, false)
+		if err != nil {
+			log.Error().Err(err).Msgf("failed to search conditation on unmounted pools")
+		}
+		log.Debug().Msgf("Found %d candidates in unmounted pools", len(candidates))
 	}
 
-	var candidates []Candidate
+	if len(candidates) == 0 {
+		return nil, pkg.ErrNotEnoughSpace{DeviceType: poolType}
+	}
+
+	sort.Slice(candidates, func(i, j int) bool {
+		// reverse sorting so most available is at beginning
+		return candidates[i].Available > candidates[j].Available
+	})
+
+	var volume filesystem.Volume
+	for _, candidate := range candidates {
+		volume, err = candidate.Pool.AddVolume(name)
+		if err != nil {
+			log.Error().Err(err).Str("pool", candidate.Pool.Name()).Msg("failed to create new filesystem")
+			continue
+		}
+		if err = volume.Limit(size); err != nil {
+			candidate.Pool.RemoveVolume(volume.Name()) // try to recover
+			log.Error().Err(err).Str("volume", volume.Path()).Msg("failed to set volume size limit")
+			continue
+		}
+
+		return volume, nil
+	}
+
+	return nil, fmt.Errorf("failed to create subvolume, logs might have more information")
+}
+
+type candidate struct {
+	Pool      filesystem.Pool
+	Available uint64
+}
+
+func (s *storageModule) checkForCandidates(size uint64, poolType pkg.DeviceType, mounted bool) ([]candidate, error) {
+	var candidates []candidate
+	for _, pool := range s.pools {
+		_, poolIsMounted := pool.Mounted()
+		if mounted != poolIsMounted {
+			continue
+		}
+		log.Debug().Msgf("checking pool %s for space", pool.Name())
```

Comment:
let's move this log line after the check of type, so we don't print that we check on a certain pool while we actually jut skip it.